src/neural-network.js

```diff
@@ -255,26 +263,66 @@ export default class NeuralNetwork {
   /**
    *
    * @param data
-   * @returns sizes
+   * Verifies network sizes are initilaized
+   * If they are not it will initialize them based off the data set.
    */
-  _getSizesFromData(data) {
-    let sizes = [];
-    let inputSize = data[0].input.length;
-    let outputSize = data[0].output.length;
-    let hiddenSizes = this.hiddenSizes;
-    if (!hiddenSizes) {
-      sizes.push(Math.max(3, Math.floor(inputSize / 2)));
+  _verifyIsInitialized(data) {
+    if (this.sizes) return;
```

Comment:
My idea was to just verify the size was set but it would probably be a good place to ensure the network was initialized and give the user feedback when they try to run without training.